tests/console/sshd.pm

```diff
@@ -67,36 +69,27 @@ sub run {
     my $changepwd = $ssh_testman . ":" . $ssh_testman_passwd;
     assert_script_run("useradd -m $ssh_testman");
     assert_script_run("echo $changepwd | chpasswd");
+    assert_script_run("usermod -aG \$(stat -c %G /dev/$serialdev) $ssh_testman");
 
-    opensusebasetest::select_serial_terminal();
-    if (is_serial_terminal()) {
-        # Make interactive SSH connection as the new user
-        type_string "ssh -v -l $ssh_testman localhost -t\n";
-        wait_serial('Are you sure you want to continue connecting \(yes/no(/\[fingerprint\])?\)\?', undef, 0, no_regex => 0);
-        type_string "yes\n";
-        wait_serial('[P|p]assword:', undef, 0, no_regex => 0);
-        type_string "$ssh_testman_passwd\n";
-        wait_serial('sshboy@', undef, 0, no_regex => 1);
-        type_string "export PS1='# '\n";
-
-        # Check that we are really in the SSH session
-        assert_script_run 'echo $SSH_TTY | grep "\/dev\/pts\/"';
-        assert_script_run 'ps ux | egrep ".* \? .* sshd\:"';
-        assert_script_run "whoami | grep $ssh_testman";
-        assert_script_run "mkdir .ssh";
-
-        # Exit properly and check we're root again
-        script_run("exit", 0);
-        assert_script_run "whoami | grep root";
-    }
-    else {
-        record_info("VirtIO N/A", "The VirtIO serial terminal is not available over here");
-        # Since we don't have the VirtIO serial terminal we need to gather public keys manually
-        assert_script_run "install -m 1700 -d ~/.ssh";
-        assert_script_run "ssh-keyscan localhost 127.0.0.1 ::1 > ~/.ssh/known_hosts";
-    }
+    # Save ~/.ssh generated in consotest_setup, keep only known_hosts to skip 'Are you sure ...' question
+    assert_script_run 'cp -a ~/.ssh ~/.ssh_bck && rm -f .ssh/id_rsa* .ssh/authorized_keys';
+
+    # avoid failures in virtio-console due to unexpected PS1
+    assert_script_run("echo \"PS1='# '\" >> ~$ssh_testman/.bashrc") if check_var('VIRTIO_CONSOLE', '1');
+
+    # Make interactive SSH connection as the new user
+    type_string "expect -c 'spawn ssh $ssh_testman\@localhost -t;expect sword:;send $ssh_testman_passwd\\r;interact'\n";
+    sleep(1);
```

Comment:
You might want to use `type_string …, wait_still_screen => 1;` instead

Reply:
wait_still_screen is virtio console ?

Reply:
same as https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10137#discussion_r421322035 applies then.

Reply:
samew as https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/10137#discussion_r421768734 applies then